### PR TITLE
Use sysctls available in macOS 12 / iOS 15 for hardware feature support.

### PR DESCRIPTION
**Pull request by kulinseth**

### CODE REVIEW

**Comment by kulinseth** [comment]
@malfet ping for review.

File: src/arm/mach/init.c [hunk 1]
```diff
@@ -336,34 +336,66 @@ void cpuinfo_arm_mach_init(void) {
...
 	}
+
 	/*
-	 * Support for ARMv8.1 Atomics & FP16 arithmetic instructions is supposed to be detected via
-	 * sysctlbyname calls with "hw.optional.armv8_1_atomics" and "hw.optional.neon_fp16" arguments
-	 * (see https://devstreaming-cdn.apple.com/videos/wwdc/2018/409t8zw7rumablsh/409/409_whats_new_in_llvm.pdf),
-	 * but on new iOS versions these calls just fail with EPERM.
-	 *
-	 * Thus, we whitelist CPUs known to support these instructions.
+	 * iOS 15 and macOS 12 added sysctls for Arm features.  Use them where
```
Comment (Maratyszcza) [inline src/arm/mach/init.c:341]: Nit: Arm->ARM

**Review by Maratyszcza — CHANGES_REQUESTED** [review]
Use lower-case for variable names, i.e. `has_feat_lse` instead of `has_FEAT_LSE`

**Review by Maratyszcza — CHANGES_REQUESTED** [review]
Please use lowercase variable names (i.e. `has_feat_lse` rather than `has_FEAT_LSE`). Other than that, the PR LGTM.

**Comment by Maratyszcza** [comment]
Thank you!

**Pull request merged**